### PR TITLE
cli: Anchor version manager (avm)

### LINKED ISSUE #586: cli: Add ability to switch between anchor CLI versions
This is important for allowing many versions of verifiable builds, because if there's a breaking change in the Anchor.toml, then the CLI won't work. If there's a way to switch between anchor CLI versions, then one can easily produce verifiable builds regardless of the Anchor.toml version being used.

### PR DESCRIPTION
**Pull request by tomlinton**
Adds the command line util `avm` and a wrapper `anchor` binary that runs the correct version of anchor-cli. Usage is explained in the docs.

Looks like book just got removed from this repo. I'll make a new PR for this documentation in the anchor-book repo. Are we deprecating the `docs/` directory or should I also add it there?

I'm not sure how to version `avm`, we might sync it with the main anchor version, or perhaps avm should live in a separate repo since tagging will get messy.

There is other approaches apart from using the wrapper binary, e.g. asking the user to set PATH to the ~/.avm/bin directory or using symlinks. The wrapper seemed like the nicest approach to me.

Closes #586.

### CODE REVIEW

**Comment by tomlinton** [comment]
Here is the commit with the [documentation](https://github.com/project-serum/anchor/pull/1385/commits/6ffd7cf4a33b79f4c09d0017ba73ac9ba203c679). I've pulled it out of this PR since the `book/` was removed.

**Comment by armaniferrante** [comment]
> since

Awesome let's add that PR to the book repo.

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ incremented for features.
 * ts: Add new `methods` namespace to the program client, introducing a more ergonomic builder API ([#1324](https://github.com/project-serum/anchor/pull/1324)).
 * ts: Add registry utility for fetching the latest verified build ([#1371](https://github.com/project-serum/anchor/pull/1371)).
 * cli: Expose the solana-test-validator --account flag in Anchor.toml via [[test.validator.account]] ([#1366](https://github.com/project-serum/anchor/pull/1366)).
+* cli: Add avm, a tool for managing anchor-cli versions ([#1385](https://github.com/project-serum/anchor/pull/1385))
```
Comment (armaniferrante) [inline CHANGELOG.md:26]:
```suggestion
* cli: Add avm, a tool for managing anchor-cli versions ([#1385](https://github.com/project-serum/anchor/pull/1385)).
```

**Comment by armaniferrante** [comment]
Should we use this in CI to install anchor? I think probably.

**Comment by paul-schaaf** [comment]
@tomlinton we are deprecating /docs yes. you can add the documentation to the seperate anchor-book repo

**Pull request merged**